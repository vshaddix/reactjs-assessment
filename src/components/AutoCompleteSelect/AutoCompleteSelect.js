import React from "react";

export default ({ onInput, onSelect, values, placeholder }) => {
  if (! values) return '';

  const options = values.map(value => <option key={value.id} value={value.id}>{value.text}</option>);

  return (
    <div>
      <div className="ui input">
        <input type="text" placeholder={`Search for a ${placeholder} ...`} onInput={(e) => onInput(e.target.value)} />
      </div>
      <select placeholder={`Select ${placeholder}`} onChange={e => onSelect(e.target.value)}>
        <option value="">Select {placeholder}</option>
        {options}
      </select>
    </div>
  );
};
