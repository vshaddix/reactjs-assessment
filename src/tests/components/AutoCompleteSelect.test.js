import React from 'react';
import { create } from "react-test-renderer";
import AutoCompleteSelect from '../../components/AutoCompleteSelect/AutoCompleteSelect';
import { shallow } from 'enzyme';

const values = [
  {
    id: 1,
    text: '1'
  },
  {
    id: 2,
    text: '2'
  }
];

it('renders (snapshot test)', () => {
  const component = create(<AutoCompleteSelect />);

  expect(component.toJSON()).toMatchSnapshot();
});

it('renders correctly when values are passed', () => {
  const component = create(<AutoCompleteSelect values={values} />);

  expect(component.toJSON()).toMatchSnapshot();
});

it('renders calls the passed onInput closure', () => {
  const onInput = jest.fn();
  const component = shallow(<AutoCompleteSelect values={values} onInput={onInput} />);
  component.find('input').simulate('input', { target: { value: 'inputtedValue' }});

  expect(onInput.mock.calls.length).toEqual(1);
});
