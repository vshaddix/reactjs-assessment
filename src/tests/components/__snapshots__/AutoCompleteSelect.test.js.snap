// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders (snapshot test) 1`] = `""`;

exports[`renders correctly when values are passed 1`] = `
<div>
  <div
    className="ui input"
  >
    <input
      onInput={[Function]}
      placeholder="Search for a undefined ..."
      type="text"
    />
  </div>
  <select
    onChange={[Function]}
    placeholder="Select undefined"
  >
    <option
      value=""
    >
      Select 
    </option>
    <option
      value={1}
    >
      1
    </option>
    <option
      value={2}
    >
      2
    </option>
  </select>
</div>
`;
